### PR TITLE
III-4483 - Add basic contact entry component

### PR DESCRIPTION
**Pull request by brampauwelyn**
### Added
- contact & reservation info entry component
Screenshot 2022-06-30 at 16 23 24

💅 Design by @simon-debruijn on [Figma](https://www.figma.com/file/dK1ggXC4D8dZ5OBNgA6AVZ/Contact-%26-Reservatie-redesign?node-id=0%3A1)

---
Ticket: https://jira.uitdatabank.be/browse/III-4483

### CODE REVIEW

File: src/pages/steps/AdditionalInformationStep.tsx [hunk 1]
```diff
@@ -209,6 +223,41 @@ const AdditionalInformationStep = ({
...
+
+  const getMergedContactAndBookingInfo = useDeepCompareMemoize<
+    () => MergedInfo | undefined
+  >(() => {
+    if (!contactInfo) return;
+    if (!bookingInfo) return contactInfo;
+
+    const emails = new Set(contactInfo['email']);
+    const urls = new Set(contactInfo['url']);
+    const phones = new Set(contactInfo['phones']);
```
Comment (geoffreydhuyvetters) [inline src/pages/steps/AdditionalInformationStep.tsx:240]: Why are we converting them to a set?

File: src/pages/steps/AdditionalInformationStep.tsx [hunk 2]
```diff
@@ -113,7 +123,10 @@ const AdditionalInformationStep = ({
 
   const [videos, setVideos] = useState([]);
 
-  const getEventByIdQuery = useGetEventByIdQuery({ id: eventId });
+  const getEventByIdQuery = useGetEventByIdQuery(
+    { id: eventId },
+    { refetchOnWindowFocus: false },
```
Comment (brampauwelyn) [inline src/pages/steps/AdditionalInformationStep.tsx:128]: We should probably remove this here?

File: src/pages/steps/AdditionalInformationStep.tsx [hunk 3]
```diff
@@ -209,6 +222,41 @@ const AdditionalInformationStep = ({
...
+      emails.add(bookingInfo['email']);
+    }
+
+    if (bookingInfo['url']) {
+      urls.add(bookingInfo['url']);
+    }
+
+    if (bookingInfo['phone']) {
+      phones.add(bookingInfo['phone']);
+    }
```
Comment (simon-debruijn) [inline src/pages/steps/AdditionalInformationStep.tsx:251]:
```suggestion
    const emails = new Set(contactInfo.email);
    const urls = new Set(contactInfo.url);
    const phones = new Set(contactInfo.phone);

    if (bookingInfo.email) {
      emails.add(bookingInfo.email);
    }

    if (bookingInfo.url) {
      urls.add(bookingInfo.url);
    }

    if (bookingInfo.phone) {
      phones.add(bookingInfo.phone);
    }
```
```suggestion
    const emails = new Set(contactInfo.email);
    const urls = new Set(contactInfo.url);
    const phones = new Set(contactInfo.phone);

    if (bookingInfo.email) {
      emails.add(bookingInfo.email);
    }

    if (bookingInfo.url) {
      urls.add(bookingInfo.url);
    }

    if (bookingInfo.phone) {
      phones.add(bookingInfo.phone);
    }
```

https://eslint.org/docs/latest/rules/dot-notation

File: src/pages/steps/AdditionalInformationStep.tsx [hunk 4]
```diff
@@ -349,6 +384,7 @@ const AdditionalInformationStep = ({
   }, [images, imageToEditId]);
 
   const invalidateEventQuery = async (field: Field) => {
+    console.log('in invalidate?', field);
```
Comment (simon-debruijn) [inline src/pages/steps/AdditionalInformationStep.tsx:387]:
```suggestion
```

File: src/pages/steps/ContactInfo.tsx [hunk 5]
```diff
@@ -90,43 +78,37 @@ const ContactInfo = ({
...
-                      },
-                    )}
+                    {mergedInfo[key].map((contactInfo, index) => {
+                      return (
+                        <Stack
+                          key={index}
+                          css={
+                            index !== 0
+                              ? css`border-top: 1px solid ${borderColorFromTheme}}`
+                              : ''
```
Comment (simon-debruijn) [inline src/pages/steps/ContactInfo.tsx:95]:
```suggestion
                            index !== 0
                              ? css`border-top: 1px solid ${borderColorFromTheme}}`
                              : css''
```

the css prop expects a type specific to styled components. `css``` translates the empty string to the correct type.

**Review by simon-debruijn — APPROVED** [review]

**Pull request merged**